### PR TITLE
make linter pluggable

### PR DESCRIPTION
**Pull request by smikes**
initial work: jslint runs but writes to stdout instead of returning data
update tests to use -L, -n to specify linter
style: remove dangling comma
spacing and intentation
rename linter output to "lint" from "eslint"
implement runner; add tests, fixtures for jslint
restore newlines
hoist helper functions
add jslint (w/version) as optional dependency

### CODE REVIEW

**Comment by hueniverse** [comment]
Isn't this a 5.2.0?

**Comment by geek** [comment]
@hueniverse yes, hadn't got to reviewing fully to make that switch.  Thanks.

**Comment by Marsup** [comment]
For what it's worth, I still believe a single switch is achievable :p

**Comment by smikes** [comment]
@marsup My impression was that `bossy` did not want the logic needed to support a single switch.  Easier for me to make 1 PR than two connected ones; if you want to resurrect the other PR, that's fine by me.

@geek I agree with this as a minor version bump, 5.2.0 would be awesome :-)

File: lib/linters/jslint/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+var formatErrors = function (error) {
+    return {
+        line: error.line,
+        severity: 'ERROR',
+        message: error.reason
+    };
+};
+
+var formatFile = function (file) {
+  return {
```
Comment (geek) [inline lib/linters/jslint/index.js]: Our styles a little different, 4 spaces and a newline after each function declaration.

**Comment by smikes** [comment]
More like that?

**Comment by geek** [comment]
@smikes perfect, thanks

I'll publish this later tonight.  Want to run through it all

**Pull request merged**